From 77c500e91db6be3f6da78946516f84749bb2f5b0 Mon Sep 17 00:00:00 2001
From: Charlie Sharpsteen <source@sharpsteen.net>
Date: Sat, 11 May 2019 16:47:01 -0700
Subject: [PATCH 10/12] Update pxp-agent to build on debian armhf

This commit updates the pxp-agent component to build on the debian-9-armhf
and debian-10-armhf platforms using Debian's cmake and to link against
Debian's Boost libraries.
---
 configs/components/pxp-agent.rb | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/configs/components/pxp-agent.rb b/configs/components/pxp-agent.rb
index 23cf900c..806f6fb8 100644
--- a/configs/components/pxp-agent.rb
+++ b/configs/components/pxp-agent.rb
@@ -33,6 +33,22 @@ component "pxp-agent" do |pkg, settings, platform|
     toolchain = ""
     special_flags += "-DCMAKE_CXX_FLAGS='#{settings[:cflags]}' -DENABLE_CXX_WERROR=OFF"
     boost_static_flag = "-DBOOST_STATIC=OFF"
+  elsif platform.is_cross_compiled_linux? && platform.name =~ /debian-(?:9|10)/
+    # Use system cmake and libraries
+    cmake = '/usr/bin/cmake'
+
+    # Env variables used by CMakeCross.txt to select the desired
+    # crossbuild-essential-<arch> toolchain.
+    pkg.environment('DEB_HOST_ARCH', platform.architecture)
+    pkg.environment('DEB_HOST_GNU_TYPE', platform.platform_triple)
+    toolchain = '-DCMAKE_TOOLCHAIN_FILE=/etc/dpkg-cross/cmake/CMakeCross.txt'
+
+    boost_static_flag = "-DBOOST_STATIC=OFF"
+    special_flags += ["-DBOOST_INCLUDEDIR=/usr/include",
+                      "-DBOOST_LIBRARYDIR=/usr/lib/#{platform.platform_triple}",
+                      "-DCMAKE_FIND_ROOT_PATH='/opt/puppetlabs/puppet;/usr'"].join(' ')
+
+    special_flags += " -DENABLE_CXX_WERROR=OFF" if platform.name =~ /debian-10/
   elsif platform.is_cross_compiled_linux?
     cmake = "/opt/pl-build-tools/bin/cmake"
     toolchain = "-DCMAKE_TOOLCHAIN_FILE=/opt/pl-build-tools/#{settings[:platform_triple]}/pl-build-toolchain.cmake"
-- 
2.20.1 (Apple Git-117)

